### ISSUE TITLE
Material compiled failed on some android device (OPPO A35, GPU PowerVR GE8320). Nead Help!

### ISSUE THREAD

**Issue by dandingol03** [google/filament#4837]
**Describe the bug**
 Failed to compile shader, error below:

> Compilation error in fragment shader "opengl_pbr":
"Compile failed.
WARNING: 0:4: Extension GL_GOOGLE_cpp_style_line_directive not supported
1 compilation warnings. \
\
"

**To Reproduce**
android device:  OPPO A35 PEFM00
filament version :  1.13.0
the scene reproduce:  start demo `sample-textured-object`

**Expected behavior**
crash

**Logs**
```
Type	Time	PID	Tag	Message

	11:09:33.744	10315	IMGSRV	:0: DoCompileShader: Failed to compile shader

	11:09:33.744	10315	Filament	Compilation error in fragment shader "opengl_pbr":
"Compile failed.
WARNING: 0:4: Extension GL_GOOGLE_cpp_style_line_directive not supported
1 compilation warnings.

"

	11:09:33.744	10315	Filament	1:   #version 300 es

	11:09:33.744	10315	Filament	2:   

	11:09:33.744	10315	Filament	3:   #define TARGET_MOBILE

	11:09:33.744	10315	Filament	4:   #extension GL_GOOGLE_cpp_style_line_directive : enable

	11:09:33.744	10315	Filament	5:   

	11:09:33.744	10315	Filament	6:   #define TARGET_GLES_ENVIRONMENT

	11:09:33.744	10315	Filament	7:   

	11:09:33.744	10315	Filament	8:   #define FILAMENT_OPENGL_SEMANTICS

	11:09:33.744	10315	Filament	9:   

	11:09:33.744	10315	Filament	10:   precision mediump float;

	11:09:33.744	10315	Filament	11:   precision mediump int;

	11:09:33.744	10315	Filament	12:   precision lowp sampler2DArray;

	11:09:33.744	10315	Filament	13:   precision lowp sampler3D;

	11:09:33.744	10315	Filament	14:   #if defined(FILAMENT_VULKAN_SEMANTICS)

	11:09:33.744	10315	Filament	15:   #define LAYOUT_LOCATION(x) layout(location = x)

	11:09:33.744	10315	Filament	16:   #else

	11:09:33.744	10315	Filament	17:   #define LAYOUT_LOCATION(x)

	11:09:33.744	10315	Filament	18:   #endif

	11:09:33.744	10315	Filament	19:   #define bool2    bvec2

	11:09:33.744	10315	Filament	20:   #define bool3    bvec3

	11:09:33.744	10315	Filament	21:   #define bool4    bvec4

	11:09:33.744	10315	Filament	22:   #define int2     ivec2

	11:09:33.744	10315	Filament	23:   #define int3     ivec3

	11:09:33.744	10315	Filament	24:   #define int4     ivec4

	11:09:33.744	10315	Filament	25:   #define uint2    uvec2

	11:09:33.744	10315	Filament	26:   #define uint3    uvec3

	11:09:33.744	10315	Filament	27:   #define uint4    uvec4

	11:09:33.744	10315	Filament	28:   #define float2   vec2

	11:09:33.744	10315	Filament	29:   #define float3   vec3

	11:09:33.744	10315	Filament	30:   #define float4   vec4

	11:09:33.744	10315	Filament	31:   #define float3x3 mat3

	11:09:33.744	10315	Filament	32:   #define float4x4 mat4

	11:09:33.745	10315	Filament	33:   struct ShadowData {

	11:09:33.745	10315	Filament	34:   highp mat4 lightFromWorldMatrix;

	11:09:33.745	10315	Filament	35:   highp vec3 direction;

	11:09:33.745	10315	Filament	36:   float normalBias;

	11:09:33.745	10315	Filament	37:   };

	11:09:33.745	10315	Filament	38:   

	11:09:33.745	10315	Filament	39:   #define FILAMENT_QUALITY_LOW    0

	11:09:33.745	10315	Filament	40:   #define FILAMENT_QUALITY_NORMAL 1

	11:09:33.745	10315	Filament	41:   #define FILAMENT_QUALITY_HIGH   2

	11:09:33.745	10315	Filament	42:   #define FILAMENT_QUALITY FILAMENT_QUALITY_LOW

	11:09:33.745	10315	Filament	43:   #define CLEAR_COAT_IOR_CHANGE

	11:09:33.745	10315	Filament	44:   #define MAX_SHADOW_CASTING_SPOTS 14

	11:09:33.745	10315	Filament	45:   #define SPECULAR_AMBIENT_OCCLUSION 0

	11:09:33.745	10315	Filament	46:   #define MULTI_BOUNCE_AMBIENT_OCCLUSION 0

	11:09:33.745	10315	Filament	47:   #define HAS_DIRECTIONAL_LIGHTING

	11:09:33.745	10315	Filament	48:   #define BLEND_MODE_OPAQUE

	11:09:33.745	10315	Filament	49:   #define POST_LIGHTING_BLEND_MODE_TRANSPARENT

	11:09:33.745	10315	Filament	50:   #define SHADING_MODEL_LIT

	11:09:33.745	10315	Filament	51:   #define MATERIAL_HAS_BASE_COLOR

	11:09:33.745	10315	Filament	52:   #define MATERIAL_HAS_ROUGHNESS

	11:09:33.745	10315	Filament	53:   #define MATERIAL_HAS_METALLIC

	11:09:33.745	10315	Filament	54:   #define MATERIAL_HAS_AMBIENT_OCCLUSION

	11:09:33.745	10315	Filament	55:   #define MATERIAL_HAS_NORMAL

	11:09:33.745	10315	Filament	56:   #define MATERIAL_NEEDS_TBN

	11:09:33.745	10315	Filament	57:   #define SHADING_INTERPOLATION 

	11:09:33.745	10315	Filament	58:   #define HAS_ATTRIBUTE_TANGENTS

	11:09:33.745	10315	Filament	59:   #define HAS_ATTRIBUTE_UV0

	11:09:33.745	10315	Filament	60:   

	11:09:33.745	10315	Filament	61:   LAYOUT_LOCATION(4) in highp vec4 vertex_worldPosition;

	11:09:33.745	10315	Filament	62:   #if defined(HAS_ATTRIBUTE_TANGENTS)

	11:09:33.745	10315	Filament	63:   LAYOUT_LOCATION(5) SHADING_INTERPOLATION in mediump vec3 vertex_worldNormal;

	11:09:33.745	10315	Filament	64:   #if defined(MATERIAL_NEEDS_TBN)

	11:09:33.745	10315	Filament	65:   LAYOUT_LOCATION(6) SHADING_INTERPOLATION in mediump vec4 vertex_worldTangent;

	11:09:33.745	10315	Filament	66:   #endif

	11:09:33.745	10315	Filament	67:   #endif

	11:09:33.745	10315	Filament	68:   LAYOUT_LOCATION(7) in highp vec4 vertex_position;

	11:09:33.745	10315	Filament	69:   #if defined(HAS_ATTRIBUTE_COLOR)

	11:09:33.745	10315	Filament	70:   LAYOUT_LOCATION(9) in mediump vec4 vertex_color;

	11:09:33.745	10315	Filament	71:   #endif

	11:09:33.745	10315	Filament	72:   #if defined(HAS_ATTRIBUTE_UV0) && !defined(HAS_ATTRIBUTE_UV1)

	11:09:33.745	10315	Filament	73:   LAYOUT_LOCATION(10) in highp vec2 vertex_uv01;

	11:09:33.745	10315	Filament	74:   #elif defined(HAS_ATTRIBUTE_UV1)

	11:09:33.745	10315	Filament	75:   LAYOUT_LOCATION(10) in highp vec4 vertex_uv01;

	11:09:33.745	10315	Filament	76:   #endif

	11:09:33.745	10315	Filament	77:   #if defined(HAS_SHADOWING) && defined(HAS_DIRECTIONAL_LIGHTING)

	11:09:33.745	10315	Filament	78:   LAYOUT_LOCATION(11) in highp vec4 vertex_lightSpacePosition;

	11:09:33.745	10315	Filament	79:   #endif

	11:09:33.745	10315	Filament	80:   

	11:09:33.745	10315	Filament	81:   layout(std140) uniform FrameUniforms {

	11:09:33.745	10315	Filament	82:       highp mat4 viewFromWorldMatrix;

	11:09:33.745	10315	Filament	83:       highp mat4 worldFromViewMatrix;

	11:09:33.745	10315	Filament	84:       highp mat4 clipFromViewMatrix;

	11:09:33.745	10315	Filament	85:       highp mat4 viewFromClipMatrix;

	11:09:33.745	10315	Filament	86:       highp mat4 clipFromWorldMatrix;

	11:09:33.745	10315	Filament	87:       highp mat4 worldFromClipMatrix;

	11:09:33.745	10315	Filament	88:       highp mat4 lightFromWorldMatrix[4];

	11:09:33.745	10315	Filament	89:       highp vec4 cascadeSplits;

	11:09:33.745	10315	Filament	90:       highp vec4 resolution;

	11:09:33.745	10315	Filament	91:       highp vec3 cameraPosition;

	11:09:33.745	10315	Filament	92:       highp float time;

	11:09:33.745	10315	Filament	93:       vec4 lightColorIntensity;

	11:09:33.745	10315	Filament	94:       vec4 sun;

	11:09:33.745	10315	Filament	95:       vec3 padding0;

	11:09:33.745	10315	Filament	96:       uint lightChannels;

	11:09:33.745	10315	Filament	97:       vec3 lightDirection;

	11:09:33.745	10315	Filament	98:       uint fParamsX;

	11:09:33.745	10315	Filament	99:       vec3 shadowBias;

	11:09:33.745	10315	Filament	100:       float oneOverFroxelDimensionY;

	11:09:33.745	10315	Filament	101:       vec4 zParams;

	11:09:33.745	10315	Filament	102:       uvec2 fParams;

	11:09:33.745	10315	Filament	103:       vec2 origin;

	11:09:33.745	10315	Filament	104:       float oneOverFroxelDimension;

	11:09:33.745	10315	Filament	105:       float iblLuminance;

	11:09:33.745	10315	Filament	106:       highp float exposure;

	11:09:33.745	10315	Filament	107:       float ev100;

	11:09:33.745	10315	Filament	108:       vec3 iblSH[9];

	11:09:33.745	10315	Filament	109:       vec4 userTime;

	11:09:33.745	10315	Filament	110:       float iblRoughnessOneLevel;

	11:09:33.745	10315	Filament	111:       float cameraFar;

	11:09:33.745	10315	Filament	112:       float refractionLodOffset;

	11:09:33.745	10315	Filament	113:       uint directionalShadows;

	11:09:33.745	10315	Filament	114:       vec3 worldOffset;

	11:09:33.745	10315	Filament	115:       float ssContactShadowDistance;

	11:09:33.745	10315	Filament	116:       float fogStart;

	11:09:33.745	10315	Filament	117:       float fogMaxOpacity;

	11:09:33.745	10315	Filament	118:       float fogHeight;

	11:09:33.745	10315	Filament	119:       float fogHeightFalloff;

	11:09:33.745	10315	Filament	120:       vec3 fogColor;

	11:09:33.745	10315	Filament	121:       float fogDensity;

	11:09:33.745	10315	Filament	122:       float fogInscatteringStart;

	11:09:33.745	10315	Filament	123:       float fogInscatteringSize;

	11:09:33.745	10315	Filament	124:       float fogColorFromIbl;

	11:09:33.746	10315	Filament	125:       uint cascades;

	11:09:33.746	10315	Filament	126:       float aoSamplingQualityAndEdgeDistance;

	11:09:33.746	10315	Filament	127:       float aoBentNormals;

	11:09:33.746	10315	Filament	128:       float aoReserved2;

	11:09:33.746	10315	Filament	129:       float aoReserved3;

	11:09:33.746	10315	Filament	130:       vec2 clipControl;

	11:09:33.746	10315	Filament	131:       vec2 padding1;

	11:09:33.746	10315	Filament	132:       float vsmExponent;

	11:09:33.746	10315	Filament	133:       float vsmDepthScale;

	11:09:33.746	10315	Filament	134:       float vsmLightBleedReduction;

	11:09:33.746	10315	Filament	135:       float vsmReserved0;

	11:09:33.746	10315	Filament	136:       float lodBias;

	11:09:33.746	10315	Filament	137:       highp float oneOverFarMinusNear;

	11:09:33.746	10315	Filament	138:       highp float nearOverFarMinusNear;

	11:09:33.746	10315	Filament	139:       float reserved3;

	11:09:33.746	10315	Filament	140:       vec4 padding2[58];

	11:09:33.746	10315	Filament	141:   } frameUniforms;

	11:09:33.746	10315	Filament	142:   

	11:09:33.746	10315	Filament	143:   layout(std140) uniform ObjectUniforms {

	11:09:33.746	10315	Filament	144:       highp mat4 worldFromModelMatrix;

	11:09:33.746	10315	Filament	145:       highp mat3 worldFromModelNormalMatrix;

	11:09:33.746	10315	Filament	146:       highp vec4 morphWeights;

	11:09:33.746	10315	Filament	147:       uint flags;

	11:09:33.746	10315	Filament	148:       uint channels;

	11:09:33.746	10315	Filament	149:       uint objectId;

	11:09:33.746	10315	Filament	150:       float userData;

	11:09:33.746	10315	Filament	151:   } objectUniforms;

	11:09:33.746	10315	Filament	152:   

	11:09:33.746	10315	Filament	153:   layout(std140) uniform LightsUniforms {

	11:09:33.746	10315	Filament	154:       highp mat4 lights[256];

	11:09:33.746	10315	Filament	155:   } lightsUniforms;

	11:09:33.746	10315	Filament	156:   

	11:09:33.746	10315	Filament	157:   layout(std140) uniform FroxelRecordUniforms {

	11:09:33.746	10315	Filament	158:       highp uvec4 records[1024];

	11:09:33.746	10315	Filament	159:   } froxelRecordUniforms;

	11:09:33.746	10315	Filament	160:   

	11:09:33.746	10315	Filament	161:   uniform mediump sampler2DArrayShadow light_shadowMap;

	11:09:33.746	10315	Filament	162:   uniform mediump usampler2D light_froxels;

	11:09:33.746	10315	Filament	163:   uniform mediump sampler2D light_iblDFG;

	11:09:33.746	10315	Filament	164:   uniform mediump samplerCube light_iblSpecular;

	11:09:33.746	10315	Filament	165:   uniform mediump sampler2DArray light_ssao;

	11:09:33.746	10315	Filament	166:   uniform mediump sampler2D light_ssr;

	11:09:33.746	10315	Filament	167:   uniform highp sampler2D light_structure;

	11:09:33.746	10315	Filament	168:   

	11:09:33.746	10315	Filament	169:   uniform  sampler2D materialParams_baseColor;

	11:09:33.746	10315	Filament	170:   uniform  sampler2D materialParams_aoRoughnessMetallic;

	11:09:33.746	10315	Filament	171:   uniform  sampler2D materialParams_normal;

	11:09:33.746	10315	Filament	172:   

	11:09:33.746	10315	Filament	173:   float filament_lodBias;

	11:09:33.746	10315	Filament	174:   

	11:09:33.746	10315	Filament	175:   #define PI                 3.14159265359

	11:09:33.746	10315	Filament	176:   #define HALF_PI            1.570796327

	11:09:33.746	10315	Filament	177:   #define MEDIUMP_FLT_MAX    65504.0

	11:09:33.746	10315	Filament	178:   #define MEDIUMP_FLT_MIN    0.00006103515625

	11:09:33.746	10315	Filament	179:   #ifdef TARGET_MOBILE

	11:09:33.746	10315	Filament	180:   #define FLT_EPS            MEDIUMP_FLT_MIN

	11:09:33.746	10315	Filament	181:   #define saturateMediump(x) min(x, MEDIUMP_FLT_MAX)

	11:09:33.746	10315	Filament	182:   #else

	11:09:33.746	10315	Filament	183:   #define FLT_EPS            1e-5

	11:09:33.746	10315	Filament	184:   #define saturateMediump(x) x

	11:09:33.746	10315	Filament	185:   #endif

	11:09:33.746	10315	Filament	186:   #define saturate(x)        clamp(x, 0.0, 1.0)

	11:09:33.746	10315	Filament	187:   #define atan2(x, y)        atan(y, x)

	11:09:33.746	10315	Filament	188:   float pow5(float x) {

	11:09:33.746	10315	Filament	189:   float x2 = x * x;

	11:09:33.746	10315	Filament	190:   return x2 * x2 * x;

	11:09:33.746	10315	Filament	191:   }

	11:09:33.746	10315	Filament	192:   float sq(float x) {

	11:09:33.746	10315	Filament	193:   return x * x;

	11:09:33.746	10315	Filament	194:   }

	11:09:33.746	10315	Filament	195:   float max3(const vec3 v) {

	11:09:33.746	10315	Filament	196:   return max(v.x, max(v.y, v.z));

	11:09:33.746	10315	Filament	197:   }

	11:09:33.746	10315	Filament	198:   float min3(const vec3 v) {

	11:09:33.746	10315	Filament	199:   return min(v.x, min(v.y, v.z));

	11:09:33.746	10315	Filament	200:   }

	11:09:33.746	10315	Filament	201:   float acosFast(float x) {

	11:09:33.746	10315	Filament	202:   float y = abs(x);

	11:09:33.746	10315	Filament	203:   float p = -0.1565827 * y + 1.570796;

	11:09:33.746	10315	Filament	204:   p *= sqrt(1.0 - y);

	11:09:33.746	10315	Filament	205:   return x >= 0.0 ? p : PI - p;

	11:09:33.746	10315	Filament	206:   }

	11:09:33.746	10315	Filament	207:   float acosFastPositive(float x) {

	11:09:33.746	10315	Filament	208:   float p = -0.1565827 * x + 1.570796;

	11:09:33.746	10315	Filament	209:   return p * sqrt(1.0 - x);

	11:09:33.746	10315	Filament	210:   }

	11:09:33.746	10315	Filament	211:   highp vec4 mulMat4x4Float3(const highp mat4 m, const highp vec3 v) {

	11:09:33.746	10315	Filament	212:   return v.x * m[0] + (v.y * m[1] + (v.z * m[2] + m[3]));

	11:09:33.746	10315	Filament	213:   }

	11:09:33.746	10315	Filament	214:   highp vec3 mulMat3x3Float3(const highp mat4 m, const highp vec3 v) {

	11:09:33.746	10315	Filament	215:   return v.x * m[0].xyz + (v.y * m[1].xyz + (v.z * m[2].xyz));

	11:09:33.746	10315	Filament	216:   }

	11:09:33.747	10315	Filament	217:   void toTangentFrame(const highp vec4 q, out highp vec3 n) {

	11:09:33.747	10315	Filament	218:   n = vec3( 0.0,  0.0,  1.0) +

	11:09:33.747	10315	Filament	219:   vec3( 2.0, -2.0, -2.0) * q.x * q.zwx +

	11:09:33.747	10315	Filament	220:   vec3( 2.0,  2.0, -2.0) * q.y * q.wzy;

	11:09:33.747	10315	Filament	221:   }

	11:09:33.747	10315	Filament	222:   void toTangentFrame(const highp vec4 q, out highp vec3 n, out highp vec3 t) {

	11:09:33.747	10315	Filament	223:   toTangentFrame(q, n);

	11:09:33.747	10315	Filament	224:   t = vec3( 1.0,  0.0,  0.0) +

	11:09:33.747	10315	Filament	225:   vec3(-2.0,  2.0, -2.0) * q.y * q.yxw +

	11:09:33.747	10315	Filament	226:   vec3(-2.0,  2.0,  2.0) * q.z * q.zwx;

	11:09:33.747	10315	Filament	227:   }

	11:09:33.747	10315	Filament	228:   

	11:09:33.747	10315	Filament	229:   #if defined(HAS_SHADOWING)

	11:09:33.747	10315	Filament	230:   #if defined(HAS_VSM)

	11:09:33.747	10315	Filament	231:   #if defined(HAS_DIRECTIONAL_LIGHTING)

	11:09:33.747	10315	Filament	232:   highp vec4 computeLightSpacePositionDirectional(const highp vec3 p, const highp vec3 n,

	11:09:33.747	10315	Filament	233:   const highp vec3 l, const float b, const highp mat4 lightFromWorldMatrix) {

	11:09:33.747	10315	Filament	234:   return mulMat4x4Float3(lightFromWorldMatrix, p);

	11:09:33.747	10315	Filament	235:   }

	11:09:33.747	10315	Filament	236:   #endif

	11:09:33.747	10315	Filament	237:   #if defined(HAS_DYNAMIC_LIGHTING)

	11:09:33.747	10315	Filament	238:   highp vec4 computeLightSpacePositionSpot(const highp vec3 p, const highp vec3 n,

	11:09:33.747	10315	Filament	239:   const highp vec3 l,  const float b, const highp mat4 lightFromWorldMatrix) {

	11:09:33.747	10315	Filament	240:   return mulMat4x4Float3(lightFromWorldMatrix, p);

	11:09:33.747	10315	Filament	241:   }

	11:09:33.747	10315	Filament	242:   #endif

	11:09:33.747	10315	Filament	243:   #else

	11:09:33.747	10315	Filament	244:   highp vec4 computeLightSpacePositionDirectional(const highp vec3 p, const highp vec3 n,

	11:09:33.747	10315	Filament	245:   const highp vec3 l, const float b, const highp mat4 lightFromWorldMatrix) {

	11:09:33.747	10315	Filament	246:   float NoL = saturate(dot(n, l));

	11:09:33.747	10315	Filament	247:   highp float sinTheta = sqrt(1.0 - NoL * NoL);

	11:09:33.747	10315	Filament	248:   highp vec3 offsetPosition = p + n * (sinTheta * b);

	11:09:33.747	10315	Filament	249:   return mulMat4x4Float3(lightFromWorldMatrix, offsetPosition);

	11:09:33.747	10315	Filament	250:   }

	11:09:33.747	10315	Filament	251:   #if defined(HAS_DYNAMIC_LIGHTING)

	11:09:33.747	10315	Filament	252:   highp vec4 computeLightSpacePositionSpot(const highp vec3 p, const highp vec3 n,

	11:09:33.747	10315	Filament	253:   const highp vec3 l, const float b, const highp mat4 lightFromWorldMatrix) {

	11:09:33.747	10315	Filament	254:   highp vec4 positionLs = mulMat4x4Float3(lightFromWorldMatrix, p);

	11:09:33.747	10315	Filament	255:   highp float oneOverZ = positionLs.w / positionLs.z;

	11:09:33.747	10315	Filament	256:   return computeLightSpacePositionDirectional(p, n, l, b * oneOverZ, lightFromWorldMatrix);

	11:09:33.747	10315	Filament	257:   }

	11:09:33.747	10315	Filament	258:   #endif

	11:09:33.747	10315	Filament	259:   #endif

	11:09:33.747	10315	Filament	260:   #endif

	11:09:33.747	10315	Filament	261:   

	11:09:33.747	10315	Filament	262:   highp mat3  shading_tangentToWorld;

	11:09:33.747	10315	Filament	263:   highp vec3  shading_position;

	11:09:33.747	10315	Filament	264:   vec3  shading_view;

	11:09:33.747	10315	Filament	265:   vec3  shading_normal;

	11:09:33.747	10315	Filament	266:   vec3  shading_geometricNormal;

	11:09:33.747	10315	Filament	267:   vec3  shading_reflected;

	11:09:33.747	10315	Filament	268:   float shading_NoV;

	11:09:33.747	10315	Filament	269:   #if defined(MATERIAL_HAS_BENT_NORMAL)

	11:09:33.747	10315	Filament	270:   vec3  shading_bentNormal;

	11:09:33.747	10315	Filament	271:   #endif

	11:09:33.747	10315	Filament	272:   #if defined(MATERIAL_HAS_CLEAR_COAT)

	11:09:33.747	10315	Filament	273:   vec3  shading_clearCoatNormal;

	11:09:33.747	10315	Filament	274:   #endif

	11:09:33.747	10315	Filament	275:   highp vec2 shading_normalizedViewportCoord;

	11:09:33.747	10315	Filament	276:   

	11:09:33.747	10315	Filament	277:   float luminance(const vec3 linear) {

	11:09:33.747	10315	Filament	278:   return dot(linear, vec3(0.2126, 0.7152, 0.0722));

	11:09:33.747	10315	Filament	279:   }

	11:09:33.747	10315	Filament	280:   float computePreExposedIntensity(const highp float intensity, const highp float exposure) {

	11:09:33.747	10315	Filament	281:   return intensity * exposure;

	11:09:33.747	10315	Filament	282:   }

	11:09:33.747	10315	Filament	283:   void unpremultiply(inout vec4 color) {

	11:09:33.747	10315	Filament	284:   color.rgb /= max(color.a, FLT_EPS);

	11:09:33.747	10315	Filament	285:   }

	11:09:33.747	10315	Filament	286:   vec3 ycbcrToRgb(float luminance, vec2 cbcr) {

	11:09:33.747	10315	Filament	287:   const mat4 ycbcrToRgbTransform = mat4(

	11:09:33.747	10315	Filament	288:   1.0000,  1.0000,  1.0000,  0.0000,

	11:09:33.747	10315	Filament	289:   0.0000, -0.3441,  1.7720,  0.0000,

	11:09:33.747	10315	Filament	290:   1.4020, -0.7141,  0.0000,  0.0000,

	11:09:33.747	10315	Filament	291:   -0.7010,  0.5291, -0.8860,  1.0000

	11:09:33.747	10315	Filament	292:   );

	11:09:33.747	10315	Filament	293:   return (ycbcrToRgbTransform * vec4(luminance, cbcr, 1.0)).rgb;

	11:09:33.747	10315	Filament	294:   }

	11:09:33.747	10315	Filament	295:   vec3 Inverse_Tonemap_Filmic(const vec3 x) {

	11:09:33.747	10315	Filament	296:   return (0.03 - 0.59 * x - sqrt(0.0009 + 1.3702 * x - 1.0127 * x * x)) / (-5.02 + 4.86 * x);

	11:09:33.747	10315	Filament	297:   }

	11:09:33.747	10315	Filament	298:   vec3 inverseTonemapSRGB(vec3 color) {

	11:09:33.747	10315	Filament	299:   color = clamp(color, 0.0, 1.0);

	11:09:33.747	10315	Filament	300:   return Inverse_Tonemap_Filmic(pow(color, vec3(2.2)));

	11:09:33.747	10315	Filament	301:   }

	11:09:33.747	10315	Filament	302:   vec3 inverseTonemap(vec3 linear) {

	11:09:33.747	10315	Filament	303:   return Inverse_Tonemap_Filmic(clamp(linear, 0.0, 1.0));

	11:09:33.748	10315	Filament	304:   }

	11:09:33.748	10315	Filament	305:   vec3 decodeRGBM(vec4 c) {

	11:09:33.748	10315	Filament	306:   c.rgb *= (c.a * 16.0);

	11:09:33.748	10315	Filament	307:   return c.rgb * c.rgb;

	11:09:33.748	10315	Filament	308:   }

	11:09:33.748	10315	Filament	309:   vec3 heatmap(float v) {

	11:09:33.748	10315	Filament	310:   vec3 r = v * 2.1 - vec3(1.8, 1.14, 0.3);

	11:09:33.748	10315	Filament	311:   return 1.0 - r * r;

	11:09:33.748	10315	Filament	312:   }

	11:09:33.748	10315	Filament	313:   vec3 uintToColorDebug(uint v) {

	11:09:33.748	10315	Filament	314:   if (v == 0u) {

	11:09:33.748	10315	Filament	315:   return vec3(0.0, 1.0, 0.0);

	11:09:33.748	10315	Filament	316:   } else if (v == 1u) {

	11:09:33.748	10315	Filament	317:   return vec3(0.0, 0.0, 1.0);

	11:09:33.748	10315	Filament	318:   } else if (v == 2u) {

	11:09:33.748	10315	Filament	319:   return vec3(1.0, 1.0, 0.0);

	11:09:33.748	10315	Filament	320:   } else if (v == 3u) {

	11:09:33.748	10315	Filament	321:   return vec3(1.0, 0.0, 0.0);

	11:09:33.748	10315	Filament	322:   } else if (v == 4u) {

	11:09:33.748	10315	Filament	323:   return vec3(1.0, 0.0, 1.0);

	11:09:33.748	10315	Filament	324:   } else if (v == 5u) {

	11:09:33.748	10315	Filament	325:   return vec3(0.0, 1.0, 1.0);

	11:09:33.748	10315	Filament	326:   }

	11:09:33.748	10315	Filament	327:   }

	11:09:33.748	10315	Filament	328:   #if defined(TARGET_MOBILE)

	11:09:33.748	10315	Filament	329:   #define MIN_PERCEPTUAL_ROUGHNESS 0.089

	11:09:33.748	10315	Filament	330:   #define MIN_ROUGHNESS            0.007921

	11:09:33.748	10315	Filament	331:   #else

	11:09:33.748	10315	Filament	332:   #define MIN_PERCEPTUAL_ROUGHNESS 0.045

	11:09:33.748	10315	Filament	333:   #define MIN_ROUGHNESS            0.002025

	11:09:33.748	10315	Filament	334:   #endif

	11:09:33.748	10315	Filament	335:   #define MIN_N_DOT_V 1e-4

	11:09:33.748	10315	Filament	336:   float clampNoV(float NoV) {

	11:09:33.748	10315	Filament	337:   return max(NoV, MIN_N_DOT_V);

	11:09:33.748	10315	Filament	338:   }

	11:09:33.748	10315	Filament	339:   vec3 computeDiffuseColor(const vec4 baseColor, float metallic) {

	11:09:33.748	10315	Filament	340:   return baseColor.rgb * (1.0 - metallic);

	11:09:33.748	10315	Filament	346:   return 0.16 * reflectance * reflectance;

	11:09:33.748	10315	Filament	371:   return saturate(f0 * (f0 * (0.941892 - 0.263008 * f0) + 0.346479) - 0.0285998);

	11:09:33.748	10315	Filament	372:   #endif

	11:09:33.749	10315	Filament	383:   }

	11:09:33.749	10315	Filament	417:   float getEV100() {

	11:09:33.749	10315	Filament	437:   return materialParams._maskThreshold;

	11:09:33.750	10315	Filament	466:   }

	11:09:33.751	10315	Filament	523:   vec4  emissive;

	11:09:33.751	10315	Filament	528:   float clearCoat;

	11:09:33.751	10315	Filament	538:   #endif

	11:09:33.751	10315	Filament	539:   #if defined(SHADING_MODEL_CLOTH)

	11:09:33.751	10315	Filament	555:   #if defined(MATERIAL_HAS_CLEAR_COAT) && defined(MATERIAL_HAS_CLEAR_COAT_NORMAL)

	11:09:33.752	10315	Filament	579:   #endif

	11:09:33.752	10315	Filament	582:   void initMaterial(out MaterialInputs material) {

	11:09:33.752	10315	Filament	599:   #endif

	11:09:33.752	10315	Filament	614:   material.subsurfaceColor = vec3(1.0);

	11:09:33.752	10315	Filament	618:   #if defined(MATERIAL_HAS_SUBSURFACE_COLOR)

	11:09:33.753	10315	Filament	637:   #endif

	11:09:33.753	10315	Filament	662:   vec3  l;

	11:09:33.753	10315	Filament	663:   float NdotL;

	11:09:33.753	10315	Filament	678:   highp vec3 n = vertex_worldNormal;

	11:09:33.754	10315	Filament	697:   shading_position = vertex_worldPosition.xyz;

	11:09:33.754	10315	Filament	704:   shading_normal = normalize(shading_tangentToWorld * material.normal);

	11:09:33.755	10315	Filament	772:           material.baseColor = texture(materialParams_baseColor, getUV0());

	11:09:33.756	10315	Filament	833:   float uThickness;

	11:09:33.756	10315	Filament	842:   

	11:09:33.756	10315	Filament	851:   #define SPECULAR_V_KELEMEN          3

	11:09:33.757	10315	Filament	875:   float oneMinusNoHSquared = 1.0 - NoH * NoH;

	11:09:33.757	10315	Filament	876:   #endif

	11:09:33.757	10315	Filament	909:   float lambdaL = NoV * length(vec3(at * ToL, ab * BoL, NoL));

	11:09:33.757	10315	Filament	912:   }

	11:09:33.758	10315	Filament	936:   return V_SmithGGXCorrelated(roughness, NoV, NoL);

	11:09:33.759	10315	Filament	1044:   #endif

	11:09:33.759	10315	Filament	1049:   vec3  F = fresnel(pixel.f0, LoH);

	11:09:33.760	10315	Filament	1084:   color *= attenuation * NoL;

	11:09:33.760	10315	Filament	1088:   (light.colorIntensity.w * light.attenuation * occlusion);

	11:09:33.760	10315	Filament	1112:   if (frameUniforms.aoSamplingQualityAndEdgeDistance > 0.0) {

	11:09:33.761	10315	Filament	1132:   depths *= -frameUniforms.cameraFar;

	11:09:33.761	10315	Filament	1133:   vec2 f = fract(cache.uv * size - 0.5);

	11:09:33.761	10315	Filament	1158:   float d  = acosFast(cosDistance);

	11:09:33.761	10315	Filament	1159:   if (min(r1, r2) <= max(r1, r2) - d) {

	11:09:33.762	10315	Filament	1207:   bn.r = dot(bnr, cache.weights);

	11:09:33.763	10315	Filament	1252:   vec3 decodeDataForIBL(const vec4 data) {

	11:09:33.763	10315	Filament	1274:   + frameUniforms.iblSH[8] * (n.x * n.x - n.y * n.y)

	11:09:33.766	10315	Filament	1451:   R[0] = vec3( c, s, 0);

	11:09:33.766	10315	Filament	1455:   float dim = float(textureSize(light_iblSpecular, 0).x);

	11:09:33.767	10315	Filament	1522:   #endif

	11:09:33.767	10315	Filament	1523:   #if defined(MATERIAL_HAS_CLEAR_COAT)

	11:09:33.767	10315	Filament	1524:   #if defined(MATERIAL_HAS_NORMAL) || defined(MATERIAL_HAS_CLEAR_COAT_NORMAL)

	11:09:33.767	10315	Filament	1525:   float clearCoatNoV = clampNoV(dot(shading_clearCoatNormal, shading_view));

	11:09:33.768	10315	Filament	1546:   #elif defined(SHADING_MODEL_CLOTH) && defined(MATERIAL_HAS_SUBSURFACE_COLOR)

	11:09:33.768	10315	Filament	1547:   Fd *= saturate(pixel.subsurfaceColor + shading_NoV);

	11:09:33.768	10315	Filament	1548:   #endif

	11:09:33.768	10315	Filament	1549:   }

	11:09:33.768	10315	Filament	1550:   #if defined(HAS_REFRACTION)

	11:09:33.768	10315	Filament	1552:   vec3 position;

	11:09:33.768	10315	Filament	1553:   vec3 direction;

	11:09:33.768	10315	Filament	1556:   void refractionSolidSphere(const PixelParams pixel,

	11:09:33.768	10315	Filament	1557:   const vec3 n, vec3 r, out Refraction ray) {

	11:09:33.768	10315	Filament	1567:   const vec3 n, vec3 r, out Refraction ray) {

	11:09:33.768	10315	Filament	1568:   vec3 rr = refract(r, n, pixel.etaIR);

	11:09:33.768	10315	Filament	1571:   ray.position = vec3(shading_position + rr * d);

	11:09:33.768	10315	Filament	1572:   ray.direction = r;

	11:09:33.768	10315	Filament	1573:   ray.d = d;

	11:09:33.768	10315	Filament	1574:   #if REFRACTION_MODE == REFRACTION_MODE_CUBEMAP

	11:09:33.768	10315	Filament	1575:   float envDistance = 10.0;

	11:09:33.768	10315	Filament	1576:   ray.direction = normalize((ray.position - shading_position) + ray.direction * envDistance);

	11:09:33.768	10315	Filament	1577:   #endif

	11:09:33.768	10315	Filament	1578:   }

	11:09:33.768	10315	Filament	1579:   void refractionThinSphere(const PixelParams pixel,

	11:09:33.768	10315	Filament	1580:   const vec3 n, vec3 r, out Refraction ray) {

	11:09:33.768	10315	Filament	1586:   ray.position = vec3(shading_position + rr * d);

	11:09:33.768	10315	Filament	1587:   #else

	11:09:33.768	10315	Filament	1588:   ray.position = vec3(shading_position);

	11:09:33.768	10315	Filament	1599:   refractionSolidSphere(pixel, n0, -shading_view, ray);

	11:09:33.769	10315	Filament	1603:   #error invalid REFRACTION_TYPE

	11:09:33.769	10315	Filament	1614:   #if REFRACTION_TYPE == REFRACTION_TYPE_THIN

	11:09:33.769	10315	Filament	1615:   E *= 1.0 + pixel.transmission * (1.0 - E.g) / (1.0 + E.g);

	11:09:33.769	10315	Filament	1616:   #endif

	11:09:33.769	10315	Filament	1627:   Ft *= 1.0 - E;

	11:09:33.769	10315	Filament	1628:   #if defined(MATERIAL_HAS_ABSORPTION)

	11:09:33.769	10315	Filament	1629:   Ft *= T;

	11:09:33.769	10315	Filament	1637:   const PixelParams pixel,

	11:09:33.769	10315	Filament	1639:   inout vec3 color) {

	11:09:33.769	10315	Filament	1644:   #endif

	11:09:33.769	10315	Filament	1645:   }

	11:09:33.769	10315	Filament	1653:   vec3 E = vec3(0.0);

	11:09:33.769	10315	Filament	1655:   #endif

	11:09:33.769	10315	Filament	1656:   SSAOInterpolationCache interpolationCache;

	11:09:33.769	10315	Filament	1657:   #if defined(BLEND_MODE_OPAQUE) || defined(BLEND_MODE_MASKED)

	11:09:33.769	10315	Filament	1658:   interpolationCache.uv = uvToRenderTargetUV(getNormalizedViewportCoord().xy);

	11:09:33.769	10315	Filament	1662:   float specularAO = specularAO(shading_NoV, diffuseAO, pixel.roughness, interpolationCache);

	11:09:33.770	10315	Filament	1663:   Fr *= singleBounceAO(specularAO) * pixel.energyCompensation;

	11:09:33.770	10315	Filament	1664:   float diffuseBRDF = singleBounceAO(diffuseAO);

	11:09:33.770	10315	Filament	1675:   #endif

	11:09:33.770	10315	Filament	1680:   evaluateSheenIBL(pixel, diffuseAO, interpolationCache, Fd, Fr);

	11:09:33.770	10315	Filament	1681:   evaluateClearCoatIBL(pixel, diffuseAO, interpolationCache, Fd, Fr);

	11:09:33.770	10315	Filament	1686:   #define SUN_AS_AREA_LIGHT

	11:09:33.770	10315	Filament	1687:   #endif

	11:09:33.770	10315	Filament	1688:   vec3 sampleSunAreaLight(const vec3 lightDirection) {

	11:09:33.770	10315	Filament	1689:   #if defined(SUN_AS_AREA_LIGHT)

	11:09:33.770	10315	Filament	1690:   if (frameUniforms.sun.w >= 0.0) {

	11:09:33.770	10315	Filament	1691:   float LoR = dot(lightDirection, shading_reflected);

	11:09:33.770	10315	Filament	1692:   float d = frameUniforms.sun.x;

	11:09:33.770	10315	Filament	1693:   highp vec3 s = shading_reflected - LoR * lightDirection;

	11:09:33.770	10315	Filament	1695:   normalize(lightDirection * d + normalize(s) * frameUniforms.sun.y) : shading_reflected;

	11:09:33.770	10315	Filament	1699:   }

	11:09:33.770	10315	Filament	1700:   Light getDirectionalLight() {

	11:09:33.770	10315	Filament	1701:   Light light;

	11:09:33.770	10315	Filament	1702:   light.colorIntensity = frameUniforms.lightColorIntensity;

	11:09:33.770	10315	Filament	1703:   light.l = sampleSunAreaLight(frameUniforms.lightDirection);

	11:09:33.770	10315	Filament	1704:   light.attenuation = 1.0;

	11:09:33.770	10315	Filament	1705:   light.NoL = saturate(dot(shading_normal, light.l));

	11:09:33.770	10315	Filament	1706:   light.channels = frameUniforms.lightChannels & 0xFFu;

	11:09:33.770	10315	Filament	1707:   return light;

	11:09:33.770	10315	Filament	1708:   }

	11:09:33.770	10315	Filament	1709:   void evaluateDirectionalLight(const MaterialInputs material,

	11:09:33.770	10315	Filament	1710:   const PixelParams pixel, inout vec3 color) {

	11:09:33.770	10315	Filament	1716:   #if defined(MATERIAL_CAN_SKIP_LIGHTING)

	11:09:33.770	10315	Filament	1719:   }

	11:09:33.771	10315	Filament	1725:   uint cascade = getShadowCascade();

	11:09:33.771	10315	Filament	1733:   if ((objectUniforms.flags & FILAMENT_OBJECT_CONTACT_SHADOWS_BIT) != 0u) {

	11:09:33.771	10315	Filament	1734:   ssContactShadowOcclusion = screenSpaceContactShadow(light.l);

	11:09:33.771	10315	Filament	1735:   }

	11:09:33.771	10315	Filament	1736:   }

	11:09:33.771	10315	Filament	1737:   visibility *= 1.0 - ssContactShadowOcclusion;

	11:09:33.771	10315	Filament	1738:   #if defined(MATERIAL_HAS_AMBIENT_OCCLUSION)

	11:09:33.771	10315	Filament	1739:   visibility *= computeMicroShadowing(light.NoL, material.ambientOcclusion);

	11:09:33.771	10315	Filament	1740:   #endif

	11:09:33.771	10315	Filament	1741:   #if defined(MATERIAL_CAN_SKIP_LIGHTING)

	11:09:33.771	10315	Filament	1745:   #endif

	11:09:33.771	10315	Filament	1746:   }

	11:09:33.771	10315	Filament	1747:   #endif

	11:09:33.771	10315	Filament	1748:   #if defined(MATERIAL_HAS_CUSTOM_SURFACE_SHADING)

	11:09:33.771	10315	Filament	1774:   }

	11:09:33.771	10315	Filament	1775:   #if defined(GEOMETRIC_SPECULAR_AA)

	11:09:33.771	10315	Filament	1779:   float variance = materialParams._specularAntiAliasingVariance * (dot(du, du) + dot(dv, dv));

	11:09:33.771	10315	Filament	1780:   float roughness = perceptualRoughnessToRoughness(perceptualRoughness);

	11:09:33.771	10315	Filament	1781:   float kernelRoughness = min(2.0 * variance, materialParams._specularAntiAliasingThreshold);

	11:09:33.771	10315	Filament	1782:   float squareRoughness = saturate(roughness * roughness + kernelRoughness);

	11:09:33.772	10315	Filament	1786:   void getCommonPixelParams(const MaterialInputs material, inout PixelParams pixel) {

	11:09:33.772	10315	Filament	1797:   #elif !defined(SHADING_MODEL_CLOTH)

	11:09:33.772	10315	Filament	1798:   pixel.diffuseColor = computeDiffuseColor(baseColor, material.metallic);

	11:09:33.772	10315	Filament	1799:   #if !defined(SHADING_MODEL_SUBSURFACE) && (!defined(MATERIAL_HAS_REFLECTANCE) && defined(MATERIAL_HAS_IOR))

	11:09:33.772	10315	Filament	1800:   float reflectance = iorToF0(max(1.0, material.ior), 1.0);

	11:09:33.772	10315	Filament	1801:   #else

	11:09:33.772	10315	Filament	1802:   float reflectance = computeDielectricF0(material.reflectance);

	11:09:33.772	10315	Filament	1803:   #endif

	11:09:33.772	10315	Filament	1817:   #else

	11:09:33.772	10315	Filament	1818:   float materialor = max(1.0, material.ior);

	11:09:33.772	10315	Filament	1819:   #endif

	11:09:33.772	10315	Filament	1820:   pixel.etaIR = airIor / materialor;

	11:09:33.772	10315	Filament	1828:   #if defined(MATERIAL_HAS_THICKNESS) || defined(MATERIAL_HAS_MICRO_THICKNESS)

	11:09:33.772	10315	Filament	1834:   pixel.absorption = vec3(0.0);

	11:09:33.772	10315	Filament	1835:   #endif

	11:09:33.772	10315	Filament	1836:   #if defined(MATERIAL_HAS_THICKNESS)

	11:09:33.772	10315	Filament	1837:   pixel.thickness = max(0.0, material.thickness);

	11:09:33.772	10315	Filament	1838:   #endif

	11:09:33.773	10315	Filament	1849:   pixel.sheenColor = material.sheenColor;

	11:09:33.773	10315	Filament	1850:   float sheenPerceptualRoughness = material.sheenRoughness;

	11:09:33.773	10315	Filament	1851:   sheenPerceptualRoughness = clamp(sheenPerceptualRoughness, MIN_PERCEPTUAL_ROUGHNESS, 1.0);

	11:09:33.773	10315	Filament	1852:   #if defined(GEOMETRIC_SPECULAR_AA)

	11:09:33.773	10315	Filament	1853:   sheenPerceptualRoughness =

	11:09:33.773	10315	Filament	1856:   pixel.sheenPerceptualRoughness = sheenPerceptualRoughness;

	11:09:33.773	10315	Filament	1861:   #if defined(MATERIAL_HAS_CLEAR_COAT)

	11:09:33.773	10315	Filament	1862:   pixel.clearCoat = material.clearCoat;

	11:09:33.773	10315	Filament	1863:   float clearCoatPerceptualRoughness = material.clearCoatRoughness;

	11:09:33.773	10315	Filament	1864:   clearCoatPerceptualRoughness =

	11:09:33.773	10315	Filament	1865:   clamp(clearCoatPerceptualRoughness, MIN_PERCEPTUAL_ROUGHNESS, 1.0);

	11:09:33.773	10315	Filament	1866:   #if defined(GEOMETRIC_SPECULAR_AA)

	11:09:33.773	10315	Filament	1867:   clearCoatPerceptualRoughness =

	11:09:33.773	10315	Filament	1878:   #if defined(SHADING_MODEL_SPECULAR_GLOSSINESS)

	11:09:33.773	10315	Filament	1879:   float perceptualRoughness = computeRoughnessFromGlossiness(material.glossiness);

	11:09:33.773	10315	Filament	1880:   #else

	11:09:33.773	10315	Filament	1881:   float perceptualRoughness = material.roughness;

	11:09:33.773	10315	Filament	1882:   #endif

	11:09:33.773	10315	Filament	1883:   pixel.perceptualRoughnessUnclamped = perceptualRoughness;

	11:09:33.773	10315	Filament	1884:   #if defined(GEOMETRIC_SPECULAR_AA)

	11:09:33.773	10315	Filament	1885:   perceptualRoughness = normalFiltering(perceptualRoughness, getWorldGeometricNormalVector());

	11:09:33.773	10315	Filament	1896:   pixel.subsurfacePower = material.subsurfacePower;

	11:09:33.773	10315	Filament	1898:   pixel.thickness = saturate(material.thickness);

	11:09:33.773	10315	Filament	1899:   #endif

	11:09:33.773	10315	Filament	1900:   }

	11:09:33.773	10315	Filament	1901:   void getAnisotropyPixelParams(const MaterialInputs material, inout PixelParams pixel) {

	11:09:33.773	10315	Filament	1902:   #if defined(MATERIAL_HAS_ANISOTROPY)

	11:09:33.773	10315	Filament	1903:   vec3 direction = material.anisotropyDirection;

	11:09:33.773	10315	Filament	1904:   pixel.anisotropy = material.anisotropy;

	11:09:33.773	10315	Filament	1905:   pixel.anisotropicT = normalize(shading_tangentToWorld * direction);

	11:09:33.774	10315	Filament	1906:   pixel.anisotropicB = normalize(cross(getWorldGeometricNormalVector(), pixel.anisotropicT));

	11:09:33.774	10315	Filament	1907:   #endif

	11:09:33.774	10315	Filament	1910:   pixel.dfg = prefilteredDFG(pixel.perceptualRoughness, shading_NoV);

	11:09:33.774	10315	Filament	1911:   #if !defined(SHADING_MODEL_CLOTH)

	11:09:33.774	10315	Filament	1912:   pixel.energyCompensation = 1.0 + pixel.f0 * (1.0 / pixel.dfg.y - 1.0);

	11:09:33.774	10315	Filament	1913:   #else

	11:09:33.774	10315	Filament	1914:   pixel.energyCompensation = vec3(1.0);

	11:09:33.774	10315	Filament	1915:   #endif

	11:09:33.774	10315	Filament	1916:   #if !defined(SHADING_MODEL_CLOTH)

	11:09:33.774	10315	Filament	1917:   #if defined(MATERIAL_HAS_SHEEN_COLOR)

	11:09:33.774	10315	Filament	1918:   pixel.sheenDFG = prefilteredDFG(pixel.sheenPerceptualRoughness, shading_NoV).z;

	11:09:33.774	10315	Filament	1919:   pixel.sheenScaling = 1.0 - max3(pixel.sheenColor) * pixel.sheenDFG;

	11:09:33.774	10315	Filament	1920:   #endif

	11:09:33.774	10315	Filament	1921:   #endif

	11:09:33.774	10315	Filament	1922:   }

	11:09:33.774	10315	Filament	1923:   void getPixelParams(const MaterialInputs material, out PixelParams pixel) {

	11:09:33.774	10315	Filament	1924:   getCommonPixelParams(material, pixel);

	11:09:33.774	10315	Filament	1925:   getSheenPixelParams(material, pixel);

	11:09:33.774	10315	Filament	1926:   getClearCoatPixelParams(material, pixel);

	11:09:33.774	10315	Filament	1927:   getRoughnessPixelParams(material, pixel);

	11:09:33.774	10315	Filament	1928:   getSubsurfacePixelParams(material, pixel);

	11:09:33.774	10315	Filament	1929:   getAnisotropyPixelParams(material, pixel);

	11:09:33.774	10315	Filament	1930:   getEnergyCompensationPixelParams(pixel);

	11:09:33.774	10315	Filament	1931:   }

	11:09:33.774	10315	Filament	1932:   vec4 evaluateLights(const MaterialInputs material) {

	11:09:33.774	10315	Filament	1933:   PixelParams pixel;

	11:09:33.774	10315	Filament	1934:   getPixelParams(material, pixel);

	11:09:33.774	10315	Filament	1935:   vec3 color = vec3(0.0);

	11:09:33.774	10315	Filament	1936:   evaluateIBL(material, pixel, color);

	11:09:33.774	10315	Filament	1945:   #endif

	11:09:33.774	10315	Filament	1951:   highp float attenuation = mix(1.0, getExposure(), emissive.w);

	11:09:33.774	10315	Filament	1953:   #endif

	11:09:33.774	10315	Filament	1957:   addEmissive(material, color);

	11:09:33.774	10315	Filament	1959:   }

	11:09:33.774	10315	Filament	1960:   layout(location = 0) out vec4 fragColor;

	11:09:33.774	10315	Filament	1961:   #if defined(MATERIAL_HAS_POST_LIGHTING_COLOR)

	11:09:33.774	10315	Filament	1962:   void blendPostLightingColor(const MaterialInputs material, inout vec4 color) {

	11:09:33.774	10315	Filament	1963:   #if defined(POST_LIGHTING_BLEND_MODE_OPAQUE)

	11:09:33.774	10315	Filament	1964:   color = material.postLightingColor;

	11:09:33.774	10315	Filament	1965:   #elif defined(POST_LIGHTING_BLEND_MODE_TRANSPARENT)

	11:09:33.775	10315	Filament	1974:   }

	11:09:33.775	10315	Filament	1978:   computeShadingParams();

	11:09:33.775	10315	Filament	1980:   initMaterial(inputs);

	11:09:33.775	10315	Filament	1981:   material(inputs);

	11:09:33.775	10315	Filament	1982:   fragColor = evaluateMaterial(inputs);

	11:09:33.775	10315	Filament	1983:   #if defined(HAS_DIRECTIONAL_LIGHTING) && defined(HAS_SHADOWING)

	11:09:33.775	10315	Filament	1984:   bool visualizeCascades = bool(frameUniforms.cascades & 0x10u);

	11:09:33.775	10315	Filament	1986:   fragColor.rgb *= uintToColorDebug(getShadowCascade());

```

**Smartphone (please complete the following information):**
 - Device: [OPPO A35 PEFM00]
 - OS: [ Android 10.0]

**mat file**
```
// Textured material

material {
    name : opengl_pbr,
    shadingModel : lit,
    parameters : [
        // Base color sRGB texture
        {
            type : sampler2d,
            name : baseColor
        },
        // Packed ambient occlusion/roughness/metallic RGB texture
        {
            type : sampler2d,
            name : aoRoughnessMetallic
        },
        // Normal map RGB texture
        {
            type : sampler2d,
            name : normal
        }
    ],
    // To sample textures our material must declare that it requires
    // a set of UV coordinates from the rendered mesh
    requires: [
        uv0
    ]
}

fragment {
    void material(inout MaterialInputs material) {
        // The normal map must be set *before* calling prepareMaterial()
        material.normal = texture(materialParams_normal, getUV0()).xyz * 2.0 - 1.0;

        prepareMaterial(material);

        material.baseColor = texture(materialParams_baseColor, getUV0());

        vec3 aoRoughnessMetallic = texture(materialParams_aoRoughnessMetallic, getUV0()).rgb;
        material.ambientOcclusion = aoRoughnessMetallic.r;
        material.roughness = aoRoughnessMetallic.g;
        material.metallic = aoRoughnessMetallic.b;
    }
}

```

**Comment by romainguy** [comment]
@bejado @pixelflinger Looks like the driver might be lying to us? We check if the extension is supported and if it isn't, we remove the directive from the shader. Somehow it doesn't work on this particular driver?

**Comment by pixelflinger** [comment]
it's just a warning though:
```
WARNING: 0:4: Extension GL_GOOGLE_cpp_style_line_directive not supported
```

I think it's another issue with the IMG compiler unfortunately.

**Comment by romainguy** [comment]
@dandingol03 Could you try with an optimized material? It usually helps the shader compilers. Make sure to not pass -g to matc.

**Comment by pixelflinger** [comment]
> @dandingol03 Could you try with an optimized material? It usually helps the shader compilers. Make sure to not pass -g to matc.

Something that has worked in the past with IMG drivers, is to use the Vulkan backend, which bypasses their compiler frontend.

**Comment by dandingol03** [comment]
> > @dandingol03 Could you try with an optimized material? It usually helps the shader compilers. Make sure to not pass -g to matc.
> 
> Something that has worked in the past with IMG drivers, is to use the Vulkan backend, which bypasses their compiler frontend.

yes, IMG drivers can run in vulkan backend.

**Comment by dandingol03** [comment]
> @dandingol03 Could you try with an optimized material? It usually helps the shader compilers. Make sure to not pass -g to matc.

Thanks. I have tried without -g flag, and the device still crash

**Comment by dandingol03** [comment]
Is There any comvenient method i can figure out the uncompatible glsl statement in the shader which can not run in the PowerVR device @romainguy @pixelflinger .  Once i replace the uncompatible statement , the device can run normally, right?

**Comment by pixelflinger** [comment]
We don't know. If you figure it out, let us know, and we'll try to work around the problem.

**Comment by dandingol03** [comment]
close first~

**Comment by pixelflinger** [comment]
Duplicate of #5118
